aspnetcore/web-api/http-repl.md

```diff
@@ -0,0 +1,875 @@
+
+```console
+https://localhost:5001/fruits~ ls
+.      [get|post]
+..     []
+{id}   [get|put|delete]
+
+https://localhost:5001/fruits~
+```
+
+Alternatively, run the `ui` command to open the web API's Swagger UI page in a browser. For example:
+
+```console
+https://localhost:5001/~ ui
+```
+
+### Navigate to an endpoint
+
+To navigate to a different endpoint on the web API, run the `cd` command:
+
+```console
+https://localhost:5001/~ cd people
+```
+
+The path following the `cd` command is case insensitive. The following output format is displayed:
+
+```console
+/people    [get|post]
+
+https://localhost:5001/people~
+```
+
+## Customize the HTTP REPL
+
+The HTTP REPL's default [colors](#set-color-preferences) can be customized. Additionally, a [default text editor](#set-the-default-text-editor) can be defined. The HTTP REPL preferences are persisted across the current session and are honored in future sessions. Once modified, the preferences are stored in the following file:
+
+# [Linux](#tab/linux)
+
+*%HOME%/.httpreplprefs*
+
+# [macOS](#tab/macos)
+
+*%HOME%/.httpreplprefs*
+
+# [Windows](#tab/windows)
+
+*%USERPROFILE%\\.httpreplprefs*
+
+---
+
+The *.httpreplprefs* file is loaded on startup and not monitored for changes at runtime. Manual modifications to the file take effect only after restarting the tool.
+
+### View the settings
+
+To view the available settings, run the `pref get` command. For example:
+
+```console
+https://localhost:5001/~ pref get
+```
+
+The preceding command displays the available key-value pairs:
+
+```console
+colors.json=Green
+colors.json.arrayBrace=BoldCyan
+colors.json.comma=BoldYellow
+colors.json.name=BoldMagenta
+colors.json.nameSeparator=BoldWhite
+colors.json.objectBrace=Cyan
+colors.protocol=BoldGreen
+colors.status=BoldYellow
+```
+
+### Set color preferences
+
+Response colorization is currently supported for JSON only. To customize the default HTTP REPL tool coloring, locate the key corresponding to the color to be changed. For instructions on how to find the keys, see the [View the settings](#view-the-settings) section. For example, change the `colors.json` key value from `Green` to `White` as follows:
+
+```console
+https://localhost:5001/people~ pref set colors.json White
+```
+
+Only the [allowed colors](https://github.com/aspnet/HttpRepl/blob/01d5c3c3373e98fe566ff5ef8a17c571de880293/src/Microsoft.Repl/ConsoleHandling/AllowedColors.cs) may be used. Subsequent HTTP requests display output with the new coloring.
+
+If `colors.json.name` doesn't have a value, `colors.json.string` is used. If `colors.json.string` doesn't have a value, `colors.json.literal` is used. If `colors.json.literal` doesn't have a value, `colors.json` is used. If `colors.json` doesn't have a value, the command shell's default text color (`AllowedColors.None`) is used.
```

Comment:
Since this is just an example of the cascading effect of the colors, should we note that somehow? Perhaps preface this section with something like (definitely needs wordsmithing): 

> The default values for the color settings will fall back to more generic settings if not specified. For instance, if `colors.json.name` ...